modApiExt/alter.lua

```diff
@@ -481,6 +481,15 @@ local function isSkill(v)
 	return type(v) == "table" and v.GetSkillEffect ~= nil
 end
 
+local function skillIndexingFn(self, key)
+	if key == "GetSkillEffect" and self.__Id then
```

Comment:
Within the indexing function, `self` refers to table within `skillIndex`, which is not actually the skill, as a result it won't have `__Id`. We could specifically store that, but at that point we might as well just store in the table/metatable a reference the skill, e.g.

```lua
skillIndex[k] = setmetatable(
  { __skill = v },
  { __index = skillIndexingFn }
)
```
We probably should also be using `rawget` here, as `self.__Id`/`self.__skill` has a chance to go through the indexing function causing infinite recursion (should only happen if its nil, but just in case)

Reply:
Alright, I fixed these issues. Ended up going with storing `id` in the metatable instead of the `skill`, since this way we also allow setting the global identifier to a different skill altogether.